lib/src/components/joystick/joystick_component.dart

```
@@ -55,8 +55,6 @@ class JoystickComponent extends JoystickController {
   void removeAction(int actionId) {
     final action = children
         .firstWhere((e) => e is JoystickAction && e.actionId == actionId);
-    if (action != null) {
-      removeChild(action);
-    }
+    removeChild(action);
```

firstWhere never returns null

---

No? What happens with no item return true for the predicate?

---

throws exception

---

We should probably write this down somewhere and handle this later

---

wdym? I think if you try to remove something that doesn't exist it should throw an error, not fail silently

---

Yeah, but iirc, this will throw a Bad element exception, which is not that obvious on what happened, maybe we could handle this and throw a more meaningful exception with a better message

---

it will throw a no element exception witch I think it's pretty clear if you called remove and provided a non-existent id

---

`IterableElementError.noElement()`